app/api/thread.py

```
@@ -404,3 +404,42 @@ async def reactivate_thread(
     if clear_cache:
         clear_cache()
     return thread_to_response(thread)
+
+
+@router.post("/{thread_id}/set-pending")
+async def set_pending_thread(
+    thread_id: int,
+    current_user: Annotated[User, Depends(get_current_user)],
+    db: AsyncSession = Depends(get_db),
```

Ruff (0.14.14)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)